### ISSUE TITLE
UQS pop up is shown on top of other screens

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#4400]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Huawei P Smart 2019
- Android version: 10
- App version: 2.13.3

## Describe the bug

The UQS pop up is shown on top of other screens, here are some examples: 
<img src="https://user-images.githubusercontent.com/67682506/142083832-bb7b4b85-e3c5-494b-80f1-bd39650300cf.jpg" width="300">
<img src="https://user-images.githubusercontent.com/67682506/142083839-b103819b-ed3d-43cc-bce3-ed50acb50d9e.png" width="300">
<img src="https://user-images.githubusercontent.com/67682506/142083840-c1c64449-4f57-4a8f-84ee-7c60aeb654c9.png" width="300">

## Steps to reproduce the issue

1. Delete and reinstall the app
2. Do not dismiss the pop up
3. Create a QR code on https://www.coronawarn.app/en/eventregistration/ and scan it with an external QR code scanner
4. Enter the app following the link
5. See the issue which is shown in the first screenshot above
6. Try the same with RA test result QR codes

## Expected behaviour

The pop up should not be shown if the app is not displaying the normal screens ("Status", "Certificates", "Check-in" or "Jounal").

---
Internal Tracking ID: [EXPOSUREAPP-10623](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10623)

**Comment by dsarkar** [comment]
Good morning @Ein-Tim, thanks for reporting: Internal Tracking ID: [EXPOSUREAPP-10623](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10623)

**Comment by fynngodau** [comment]
Thanks for reporting, Tim!

**Comment by dsarkar** [comment]
FYI PR https://github.com/corona-warn-app/cwa-app-android/pull/4607

**Comment by MikeMcC399** [comment]
@Ein-Tim 

Did you look at this again? 

Regarding your steps to reproduce: with the 2.17.2 version I did not see the pop-up appear after resetting the app, only after reinstalling.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

I currently don't have access to the Android device anymore, thus I can't check, sorry.

Maybe @fynngodau would like to test this again?

**Comment by MikeMcC399** [comment]
@Ein-Tim 
I tried on 2.18.1 and again, the pop-up does not appear after resetting the app.
I simulated a re-install by clearing the app's data and I then scanned an event code. This took me to the onboarding screen "LET'S GET STARTED". After completing onboarding, the QR Code Scanner pop-up was shown on top of the Check In screen. 

This means that the issue is not resolved.

It may be that this is considered acceptable behaviour though. Normally the pop-up would be shown once only after installation. It would then be dismissed by the user and never shown again.

**Comment by MikeMcC399** [comment]
- I suggested in the new issue https://github.com/corona-warn-app/cwa-app-android/issues/4940 to remove the UQS pop-up. I don't think that this pop-up is needed any more. It was useful, when version 2.12 was released, since previously there were different places to scan QR codes depending on their contents. The Universal QR code scanner allows all CWA-related QR codes to be scanned from one button. Version 2.12 was released in October 2021 and by this time users should be accustomed to the functionality and should therefore not need a special one-time pop-up to remind them.

**Comment by MikeMcC399** [comment]
@dsarkar 

Either this issue should be closed (and a new issue opened to cover the remaining issue) or the label https://github.com/corona-warn-app/cwa-app-android/labels/Fix%202%2E16 should be removed.

**Comment by dsarkar** [comment]
@Ein-Tim see comment https://github.com/corona-warn-app/cwa-app-android/issues/4400#issuecomment-1117104473. Following @MikeMcC399's suggestion, do you want to close this issue since the remaining issue is covered in https://github.com/corona-warn-app/cwa-app-android/issues/4940?

**Comment by Ein-Tim** [comment]
This issue was never fixed, especially problematic is when the pop up is shown on top of screens although the UQS icon is not visible.

I'm unsure if I should close this issue although it was not addressed. I think it should stay open until either @MikeMcC399's enhancement request us implemented or the bug fixed.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

If you want to leave this issue open, then I suggest you update the steps to reproduce, because in current versions, resetting the app does not cause the pop-up to re-appear.

**Comment by Ein-Tim** [comment]
Done @MikeMcC399

**Comment by MikeMcC399** [comment]
@Ein-Tim 

Thanks for updating the steps to reproduce. The app is unusable unless the "QR Code Scanner" pop-up is dismissed, so it is quite intrusive. Dismissing the pop-up is however quite a simple operation (tap on "X").

If this issue is to stay open, then the labels https://github.com/corona-warn-app/cwa-app-android/labels/Fix%202%2E16 and https://github.com/corona-warn-app/cwa-app-android/labels/ready-to-close should be removed.

**Comment by Ein-Tim** [comment]
@dsarkar Please remove the labels as @MikeMcC399 suggested above.

**Comment by svengabr** [comment]
I did re-open the closed Jira ticket EXPOSUREAPP-10623

**Comment by dsarkar** [comment]
FYI PR https://github.com/corona-warn-app/cwa-app-android/pull/5366

**Comment by Ein-Tim** [comment]
@dsarkar Are you sure that this PR prevents the showing of the UQS tool tip on top of other screens? I'm only seeing changes in regard to the tap area in the PR.

**Comment by dsarkar** [comment]
@Ein-Tim You are right. PR is still interesting here since a related topic.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

- PR https://github.com/corona-warn-app/cwa-app-android/pull/5366 has been changed to stop the UQS tooltip appearing on a new install, so your steps to reproduce in the OP will no longer show the issue.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

- You might like to try out [release/2.26.x](https://github.com/corona-warn-app/cwa-app-android/tree/release/2.26.x) since the reworked PR #5366 has been merged. Perhaps your expectations are now met in order to resolve this issue?

**Comment by Ein-Tim** [comment]
@MikeMcC399 

I currently don't use Android Studio, but as I wanted to begin with it anyways, I will probably take this chance and take a look at it as soon as I have time.

**Comment by mtwalli** [comment]
@Ein-Tim I checked it this in 2.26 and can't reproduce, probably because of the changes made by https://github.com/corona-warn-app/cwa-app-android/pull/5366.

**Comment by Ein-Tim** [comment]
I currently don't have access to my Android device, but as soon as I get back home I'll test if this bug is fixed successfully with version 2.26.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

- If you delete and re-install the app, then no UQS pop-up is shown with CWA Android 2.26.
- I already closed the related issue https://github.com/corona-warn-app/cwa-app-android/issues/4940.

**Comment by Ein-Tim** [comment]
@MikeMcC399 Okay, thanks! This issue seems fixed so I'm closing it now.
